R/util_assert.R

```diff
@@ -171,3 +171,30 @@ assert_names_setequal <- function(x, expected, name = deparse(substitute(x))) {
 
   invisible(x)
 }
+
+
+assert_increasing <- function(x, strict = TRUE, name = deparse(substitute(x))) {
+  dx <- diff(x)
+  if (strict) {
+    if (any(dx <= 0)) {
+      stop(sprintf("'%s' must be strictly increasing", name))
+    }
+  } else {
+    if (any(dx < 0)) {
+      stop(sprintf("'%s' must be increasing (or equal)", name))
+    }
+  }
+  invisible(x)
+}
+
+
+assert_date_string <- function(x, name = deparse(substitute(x))) {
+  if (inherits(x, "Date")) {
+    x <- as.character(x)
+  }
+  if (!all(grepl("^[0-9]{4}-[0-9]{1,2}-[0-9]{1,2}$", x))) {
```

Comment:
note that this will allow dates like `2021-20-99`; if you *really* want to validate, we also need to do:

```
err <- is.na(suppressWarnings(as.Date(x)))
if (any(err)) {
  stop("Date format is invalid")
}
```

but we don't do this in sircovid yet anyway...